lib/connectors/registry.rb

```diff
@@ -4,49 +4,53 @@
 # you may not use this file except in compliance with the Elastic License.
 #
 
+require 'active_support/core_ext/class/subclasses'
+require 'connectors'
+
 module Connectors
-  class Factory
-    attr_reader :connectors
+  class Registry
```

Comment:
I'm not sure about making it static class - we already have problems with singletons and module methods, I think it would be better to keep this change out of PR and discuss it better.

IMO in the end the best we could do is get rid of `Connectors::REGISTRY` and just use one instance of `Connectors::Factory` that's instantiated by App. It's a class that is not used a lot and it can be trivial to pass it around to the places that use it.